### ISSUE TITLE
Type-hinting against FilesystemInterface

### ISSUE THREAD

**Issue by reinink** [thephpleague/flysystem#359]
Hi Frank, I'm just running into a small little DocBlocks issue, and was hoping for your help. In my library I'm type-hinting against `FilesystemInterface`, but I'm getting a Scrutinizer [error](https://scrutinizer-ci.com/g/reinink/glide/inspections/3150bc09-c63f-432a-91c8-e61719ca666e/issues/files/src/Server.php?status=new&orderField=path&order=asc) on the `ReadInterface` DobBlocks. They indicate that this method returns `false|array`, but the implementation actually returns a `string`.

```php
    /**
     * Read a file
     *
     * @param   string  $path
     * @return  false|array
     */
    public function read($path);
```

Am I missing something, or should [this](https://github.com/thephpleague/flysystem/blob/master/src/ReadInterface.php#L21) be `false|string` instead?

**Comment by frankdejonge** [comment]
@reinink this is fixed in the dev master version which will be released as 1.0.0 this month. The API is stable now, only the adapters are pending a move to separate repo's.

**Comment by reinink** [comment]
Good stuff!
